### ISSUE TITLE
backslash added to metabolite names

### ISSUE THREAD

**Issue by tabbassidaloii** [sec2pri/mapping_preprocessing#23]
@DeniseSl22 
In the metabolite files when there is `'` or `"` in the metabolite name a `\` has been added, see the example below:

Q425039	"bromcresol purple"	"5',5\"-dibromo-o-cresolsulfophthalein"

**Comment by DeniseSl22** [comment]
Yes, chemical names can be difficult..... But this is the actual name in the Wikidata entry:
https://www.wikidata.org/wiki/Q425039

I believe the backslash is there, so that R can read the data, right? Or is that not working?

**Comment by DeniseSl22** [comment]
I've now changes the entry, so the name is 5',5'' (so using the single quote twice, iso using a double quote).
That might resolve things for this entry; I'll run the Wikidata GitHub action again. Any others which are causing issues @tabbassidaloii ?

**Comment by tabbassidaloii** [comment]
It messes up in R when there is a `/`
It reads with no issue when I remove the unnecessary `/` manually.
Also when there is a `/` in the name, an extra `/` would be added.

**Comment by DeniseSl22** [comment]
Forward or backward slash, which one is causing the issue?

**Comment by tabbassidaloii** [comment]
backslashes are added when there is `'`, `"`, or a backslash (`/`).

**Comment by tabbassidaloii** [comment]
I remove them manually in the file, to keep the app run correctly for now.

**Comment by DeniseSl22** [comment]
How many are there, could you give me some more examples?
Then I can (try) to construct a regex to solve this.... sing TSV should already solve issues around using a comma within a name...

**Comment by tabbassidaloii** [comment]
it is around 20 to 30, I think.
Here are some more examples:
```
5',5\\"-dibromo-o-cresolsulfophthalein
Germa-Medica \\"Mg\\"
2,5-O,O-BIS-(3',3\\"-AMIDINOPHENYL)-1,4:3,6-DIANHYDRO-D-SORBITOL
diinosine-5',5\\"-pentaphosphate
3-O-beta-D-glucopyranosylpresenegenin 28-O-{[beta-D-apiofuranosyl(1-3)][beta-D-galactopyranosyl(1-4)-beta-D-xylopyranosyl(1-4)]-alpha-L-rhamnopyranosyl(1-3)}{4-O-(E)-4\\"-methoxycinnamoyl}-beta-D-fucopyranoside
3-O-beta-D-glucopyranosylpresenegenin 28-O-{[beta-D-apiofuranosyl(1-3)][beta-D-galactopyranosyl(1-4)-beta-D-xylopyranosyl(1-4)]-alpha-L-rhamnopyranosyl(1-3)}{4-O-(Z)-4\\"-methoxycinnamoyl}-beta-D-fucopyranosid
alpha-GalCer-6\\"-(pyridin-4-yl)carbamate
alpha-GalCer-6\\"-(4-pyridyl)carbamate
2'-[2\\"-(5'''-phosphoribosyl)-5\\"-phosphoribosyl]adenosine 5'-monophosphate
2'-[2\\"-(1'''-ribosyl)-1\\"-ribosyl]adenosine 5',5\\",5'''-tris(phosphate)
alpha-GalCer-6\\"(1-naphthyl)carbamate
2'-(5\\"-phosphoribosyl)adenosine 5'-monophosphate
2'-(1\\"-ribosyl)adenosine 5',5\\"-bis(phosphate)
alpha-GalCer-6\\"-(4-chlorophenyl)carbamate
N,N',N\\"-trimethyl-1,4,7-triazacyclononane
12,24-Dihydro-5H-naphtho[2,3-h]naphth[2\\",3\\":6,7]anthra[2,1,9-mna]acridine-5,10,13,18,25-pentone
```

**Comment by tabbassidaloii** [comment]
@DeniseSl22 
I don't understand why you filter the results. We should avoid adding the extra backslashes that are added.
The backslashes are in the output of the queries (and I am not sure how to avoid it) so I would try to solve it before reading the file in R.
for example,  `Germa-Medica "Mg"` in the output of query instead of `Germa-Medica \\"Mg\\"`.

There are options in R to remove them, but I have concerns about causing unnecessary changes in the other values (e.g. there are metabolites with a backslash (`/`) in their names, and we should keep those backslashes). So I would fix it in bash script.

**Comment by tabbassidaloii** [comment]
This is weird because I see it in the files I downloaded from GitHub, before opening them in R.
Yeah, let's meet either at 9 or at 11 am.

**Comment by DeniseSl22** [comment]
As discussed:
- [ ] Not push Wikidata data into Github, since this added backslashes which doesn't work in R for reading the data.
- [ ] Create a logfile, counting the lines for each file.
- [ ] Compare logfile of previous release to new one, and print out results to terminal.

**Comment by tabbassidaloii** [comment]
@DeniseSl22 if #9 is solved we can work on [#106](https://github.com/sec2pri/mapping_preprocessing/issues/106) considering below
> As discussed:
> 
> * [ ]  Not push Wikidata data into Github, since this added backslashes which doesn't work in R for reading the data.
> * [ ]  Create a logfile, counting the lines or (unique) wordcount for each file (bash), push logfile to Github.
> * [ ]  Compare logfile of previous release to new one, and print out results to terminal (bash)

**Comment by DeniseSl22** [comment]
Check the output of Wikidata queries for special characters! @DeniseSl22

**Comment by DeniseSl22** [comment]
In the current output, there are still some chemicals with a '\' in their name (I only found 2 now):
```
Q74410749		"(2S)-2-ammonio-4-\{[(1S)-1-carboxy-2-(1H-imidazol-4-yl)ethyl]ammonio\}butanoate"	
Q74417452		"(2S)-2-ammonio-4-\{[(1R)-1-carboxy-2-(1H-imidazol-4-yl)ethyl]ammonio\}butanoate"

```

Forward slashes are more common (in names, synonyms):

```
Q7531599	Q48566829	"sitagliptin/metformin"	"Metformin / sitagliptin"
Q7531599	Q48566829	"sitagliptin/metformin"	"sitagliptin / metformin hydrochloride"
Q7531599	Q48566829	"sitagliptin/metformin"	"Sitagliptin Phosphate, Metformin Hydrochloride Drug Combination"
Q7597226		"carbidopa / levodopa / entacapone"	"Corbilta (previously Levodopa/Carbidopa/Entacapone Sandoz)"
Q7597226		"carbidopa / levodopa / entacapone"	"levodopa / carbidopa / entacapone"

```

@tabbassidaloii : will this cause issues loading the data in R (you can download the query results [here](https://github.com/sec2pri/mapping_preprocessing/actions/runs/14215012483/artifacts/2865598771)), and if yes, what do you want us to do about it? I can detect the slashes (or other special characters), add them a separate file (so we can fix them manually in Wikidata), and remove them from the file used for the Shiny app?
